types/node/index.d.ts

```diff
@@ -838,7 +848,13 @@ declare module "http" {
         end(str: string, encoding?: string, cb?: Function): void;
         end(data?: any, encoding?: string): void;
     }
-    export interface ClientRequest extends stream.Writable {
+
+    // https://github.com/nodejs/node/blob/master/lib/_http_client.js#L77
+    export class ClientRequest extends stream.Writable {
+        constructor(url: string, cb?: Function);
```

Comment:
Consider to combine the constructors by using union types.
Consider also to use a stricter typing for the cb (e.a. ` (res: IncomingMessage) => void`).